engine/src/main/java/org/hibernate/search/engine/impl/AnalyzerReferenceRegistry.java

```diff
@@ -82,11 +80,59 @@ public T getAnalyzerReference(Class<?> analyzerClazz) {
 	}
 
 	public void initialize(Map<String, AnalyzerDef> analyzerDefinitions) {
-		strategy.initializeNamedAnalyzerReferences( referencesByName, analyzerDefinitions );
+		List<AnalyzerReference> references = new ArrayList<>();
```

Comment:
This looks like fine after checking other parts but I found it a bit suspicious:

you seem to put some special effort to collect all references into a single collection, to then invoke the initialize method on all references at once.
I found that suspicious, as relying on that would mean we could have some issues with dynamic discovery of new entities (and hence new analyzers). Still, it doesn't look like the initialization method actually requires to have them all in one single collection.